### PR TITLE
Alert Migration for different LAWS

### PR DESCRIPTION
**Pull request by suhanishri**
1. This PR contains alert migration when the LAWS of AMSv1 and v2 are different.
2. We get the list of Alerts using the 2018 API version of scheduled queries as AMSv1 had their queries written using the same version.

### CODE REVIEW

File: src/Migration.ps1 [hunk 1]
```diff
@@ -1,12 +1,12 @@
 ﻿param(
 #[Parameter(Mandatory=$true)]
-[string]$providerType = $null,
+[string]$providerType = "saphana",
 
 #[Parameter(Mandatory=$true)]
-[string]$amsv1ArmId = "<amsv1-arm-id>",
+[string]$amsv1ArmId = "/subscriptions/49d64d54-e966-4c46-a868-1999802b762c/resourceGroups/rg-ams-migration-test/providers/Microsoft.HanaOnAzure/sapMonitors/ams-v1-migration-full",
```
Comment (mohitmakhija1) [inline src/Migration.ps1]: remove the id's

File: src/Migration.ps1 [hunk 2]
```diff
@@ -57,7 +58,7 @@ function Main
 	}
 
     $logger.LogInfo("Please select an account to connect to Azure Portal...")
-    Connect-AzAccount -UseDeviceAuthentication;
+    #Connect-AzAccount -UseDeviceAuthentication;
```
Comment (mohitmakhija1) [inline src/Migration.ps1]: un-comment this

**Review by mohitmakhija1 — CHANGES_REQUESTED** [review]

File: src/Migration.ps1 (hunk 1 above)
Comment (suhanishri) [reply src/Migration.ps1]: Done

File: src/Migration.ps1 (hunk 2 above)
Comment (suhanishri) [reply src/Migration.ps1]: Done

File: src/AmsOperationsHelper.ps1 [hunk 3]
```diff
@@ -288,6 +341,57 @@ function GetAmsV2ManagedFunc([string]$subscriptionId, [string]$resourceGroup, [s
...
+    $rawToken = Get-AzAccessToken -ResourceTypeName Arm
+    $armToken = $rawToken.Token
+	$v2ApiVersion = "2020-02-07-preview";
+
+    $headers = @{
+        "Content-Type" = "application/json"
+        "Authorization" = "Bearer $armToken"
+    }
+
+    [string]$url = $url = "https://management.azure.com/"
```
Comment (amitms2103) [inline src/AmsOperationsHelper.ps1]: the second $url is needed ?

File: src/KeyvaultHelperFunctions.ps1 [hunk 4]
```diff
@@ -109,7 +109,7 @@ DeleteAndPurgeSecretFromKeyVault -keyVaultName $keyVaultName -secretKey $name -l
 #>
 function DeleteAndPurgeSecretFromKeyVault([string]$keyVaultName, [string]$secretKey, $logger) {
 	try {
-		Add-KeyVaultRoleAssignment -keyVaultName $keyVaultName -logger $logger;
+		# Add-KeyVaultRoleAssignment -keyVaultName $keyVaultName -logger $logger;
```
Comment (amitms2103) [inline src/KeyvaultHelperFunctions.ps1]: Remove the line if not needed

File: src/Migration.ps1 [hunk 5]
```diff
@@ -214,18 +228,77 @@ function Main
...
+	{
+		$logger.LogInfo("AMSv1 $($compareLaws.amsv1LawsId) and AMSv2 $($compareLaws.amsv2LawsId) LAWS are NOT equal..");
+		[string]$dialogContent = "Please select an action for Alert Migration.";
+		Write-Host $dialogContent;
+		while (($isMigrateAlerts -ne "yes") -and ($isMigrateAlerts -ne "no")) {
+			$isMigrateAlerts = Read-Host -Prompt "Do you wish to Migrate Alerts? (yes/no)";
+		}
+	}
+
+	if($isMigrateAlerts -like "yes") {
```
Comment (amitms2103) [inline src/Migration.ps1:248]: So for no there is no particular case to handle, only this if check for "yes" and the corresponding action suffices ?

File: src/KeyvaultHelperFunctions.ps1 (hunk 4 above)
Comment (suhanishri) [reply src/KeyvaultHelperFunctions.ps1]: Tagging @M2skills for more infor. Should we remove it?

File: src/AmsOperationsHelper.ps1 (hunk 3 above)
Comment (suhanishri) [reply src/AmsOperationsHelper.ps1]: removed.

File: src/KeyvaultHelperFunctions.ps1 (hunk 4 above)
Comment (mohitmakhija1) [reply src/KeyvaultHelperFunctions.ps1]: yes, remove this as we are providing access at the beginning of the main function

Comment (suhanishri) [reply src/KeyvaultHelperFunctions.ps1]: removed

File: src/Migration.ps1 (hunk 5 above)
Comment (suhanishri) [reply src/Migration.ps1:248]: maybe we can add a log. other than that no action required i believe

Comment (amitms2103) [reply src/Migration.ps1:248]: No need for logs, I was just clarifying.

**Review by amitms2103 — DISMISSED** [review]

File: src/Migration.ps1 (hunk 5 above)
Comment (suhanishri) [reply src/Migration.ps1:248]: Updated small log for info.

**Review by amitms2103 — APPROVED** [review]

**Review by mohitmakhija1 — APPROVED** [review]

**Pull request merged**